### ISSUE TITLE
Consumer Idle Heartbeats

### ISSUE THREAD

**Issue by ripienaar** [nats-io/nats-architecture-and-design#5]
## Overview

Consumer heartbeats can help clients to detect when an ephemeral consumer have been lost due to network interruption or if messages were lost.

The behavior is documented in ADR-9.

## Clients and Tools

 - [x] Schemas @ripienaar
 - [x] CLI @ripienaar
 - [x] Terraform @ripienaar
 - [x] Github Actions @ripienaar
 - [ ] Go @wallyqs
 - [ ] Java @scottf
 - [ ] JavaScript @aricart
 - [ ] .Net @ColinSullivan1
 - [ ] C @kozlovic
 - [ ] Python @wallyqs
 - [ ] Rust @spacejam
   
## Other Tasks

 - [ ] docs.nats.io updated
 - [ ] Update ADR to `Implemented`
 - [ ] Update client features spreadsheet

Client authors please update with your progress. If you open issues in your own repositories as a result of this request, please link them to this one by pasting the issue URL in a comment or main issue description.

**Comment by kozlovic** [comment]
@ripienaar FYI: reference to ADR-9 points to page not found. I don't think there is such ADR at the moment?

**Comment by ripienaar** [comment]
> @ripienaar FYI: reference to [ADR-9](https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-9.md) points to page not found. I don't think there is such ADR at the moment?

fixed :)

**Comment by kozlovic** [comment]
there seem to be adr-9 in the nats-server repo, so is it just that they have not been moved around yet?

**Comment by ripienaar** [comment]
> there seem to be [adr-9](https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-9.md) in the nats-server repo, so is it just that they have not been moved around yet?

Yeah there's some confusion, @aricart left a gap for me while my KV one is being reviewed and I confused things moving them here.  let me fix that

**Comment by aricart** [comment]
@ripienaar the javascript client in its current incarnation doesn't 'handle' the error, but enables clients to handle it - if the current box is simply whether you can solicit the heart beating then this is handled, however as an automated solution is likely more desirable.
